tools/marvin/marvin/lib/base.py

```diff
@@ -3876,7 +3876,12 @@ def add(cls, apiclient, services, physicalnetworkid,
         else:
             cmd.transportzoneuuid = services['transportZoneUuid']
 
-        return NiciraNvp(apiclient.addNiciraNvpDevice(cmd).__dict__)
+        if l2gatewayserviceuuid:
+            cmd.l2gatewayserviceuuid = l2gatewayserviceuuid
+        else:
+            cmd.l2gatewayserviceuuid = services['l2gatewayserviceuuid']
```

Comment:
There should be the possibility to create the device without the l2 gateway uuid. This construct forces one to exists. Please make sure you test of `services` has key `'l2gatewayserviceuuid'`before you retrieve it.